### PR TITLE
Add a bot that generates branch-to-branch merges

### PR DESCRIPTION
**Pull request by TyOverby**
Two things that are keeping this PR from being ready to review: 

* I haven't reviewed it yet and it was the result of a series of hacks.
* I haven't changed the target repository or branch yet.  Next time we need to do a master -> future merge, I'll change it.

### CODE REVIEW

**Comment by TyOverby** [comment]
test prtest/mac/dbg/unit32 please

**Comment by TyOverby** [comment]
@dotnet/roslyn-infrastructure Please review.

**Comment by TyOverby** [comment]
[An example PR generated by this bot](https://github.com/dotnet/roslyn/pull/7341)

**Comment by jmarolf** [comment]
:+1:

**Comment by TyOverby** [comment]
test prtest/mac/dbg/unit32 please

@agocke - Please review

**Comment by TyOverby** [comment]
@dotnet/roslyn-infrastructure 

Ping on review.

File: src/Tools/Github/GitMergeBot/Program.cs [hunk 1]
```diff
@@ -0,0 +1,122 @@
+﻿using Octokit;
```
Comment (brettfo) [inline src/Tools/Github/GitMergeBot/Program.cs]: Copyright

**Comment by brettfo** [comment]
:shipit:

**Pull request merged**